src/cfml/system/util/CompileDSL.cfc

```diff
@@ -0,0 +1,402 @@
+	}
+
+	function fromSource( required any sourcePaths ) {
+		if( isSimpleValue( arguments.sourcePaths ) ) {
+			arguments.sourcePaths = listToArray( arguments.sourcePaths, ",", true );
+		}
+		arguments.sourcePaths = arguments.sourcePaths.map( function( s ) {
+			return fileSystemutil.resolvePath( arguments.s, getProjectRoot() );
+		} );
+		variables.sourcePaths = arguments.sourcePaths;
+		return this;
+	}
+
+    function toClasses( required classOutputDirectory ){
+        setClassOutputDirectory( fileSystemutil.resolvePath( classOutputDirectory, getProjectRoot() ) )
+        return this;
+    }
+
+    function withVerbose() {
+        setVerbose( true );
+        return this;
+    }
+
+    function withEncoding( required encodeValue ) {
+        setEncode( encodeValue );
+        return this;
+    }
+
+	function recursive() {
+		setRecursive( true );
+		return this;
+	}
+
+	function toJar( string jarName='' ) {
+		setJarNameString( jarName );
+		setCreateJar( true );
+		return this;
+	}
+
+	function libsDir( required libsDir ) {
+		setLibsDir(libsDir);
+		return this;
+	}
+
+	function compileOptions( required string options ) {
+		setCompileOptionsString( options );
+		return this;
+	}
+
+	function jarOptions( required string options) {
+		setJarOptionsString( options );
+		return this;
+	}
+
+    function run() {
+		job.start( 'Compiling' );
+
+        job.start( 'find jdk bin directory' );
+		setJavaBinFolder( findJDKBinDirectory() );
+        job.complete( getVerbose() );
+
+        job.start( 'compiling the code' );
+		compileCode();
+        job.complete( getVerbose() );
+
+		if( getCreateJar() ) {
+            job.start( 'creating the jar' );
+    		buildJar();
+            job.complete( getVerbose() );
+		}
+
+		job.complete( getVerbose() );
+    }
+
+	function compileCode() {
+
+		if( directoryExists( getClassOutputDirectory() ) ){
+			directoryDelete( getClassOutputDirectory(), true )
+		}
+
+		//shell.printString( " glob-> start... " );
+        //job.addLog( " glob-> start... " );
+
+		var globber = wirebox.getInstance( 'globber' );
+		var tempSrcFileName = tempDir & 'temp#createUUID()#.txt';
+
+		setSourcePaths( getSourcePaths().map( function( p ) {
+			var currentPath = fileSystemutil.resolvePath( arguments.p, getProjectRoot() );
+			if ( directoryExists( currentPath ) ) {
+				currentPath &= "**.java";
+				return currentPath;
+
+			} else if ( FileExists( currentPath ) ) {
+				return currentPath;
+
+			} else if ( Find("**", currentPath) ) {
+				return currentPath;
+
+			} else {
+				throw(
+					message='Non-Existing Folder', detail=currentPath & ' does not exist',
+					type="commandException"
+				);
+
+			}
+			return currentPath;
+
+		} ) );
+
+        job.addLog( " " & serialize( getSourcePaths() ) & " " );
+		try{
+
+			writeTempSourceFile( tempSrcFileName );
+
+			var javacCommand = 'run "#getJavaBinFolder()#javac" "@#tempSrcFileName#" -d #variables.classOutputDirectory# #variables.compileOptionsString#';
```

Comment:
Need to account for quotes in `#variables.classOutputDirectory#`

Reply:
@bdw429s do you mean "#variables.classOutputDirectory#" ?

Reply:
You tell me.  How do you handle a file path argument with spaces in it at the command line? :)